src/main/java/org/broadinstitute/hellbender/tools/copynumber/PostprocessGermlineCNVCalls.java

```diff
@@ -325,6 +347,48 @@ private void generateIntervalsVCFFileFromAllShards() {
         intervalsVCFWriter.close();
     }
 
+    private void concatenateDenoisedCopyRatioFiles() {
+        if (outputDenoisedCopyRatioFile != null) {
+            logger.info("Concatenating the denoised copy ratios...");
+            final List<SimpleInterval> concatenatedIntervalList = new ArrayList<>();
+            final List<LinearNonLocatableCopyRatio> concatenatedDenoisedCopyRatioRecordsList = new ArrayList<>();
+            /* Read in and concatenate all denoised copy ratio files into one list */
+            for (int shardIndex = 0; shardIndex < numShards; shardIndex++) {
+                final File shardRootDirectory = sortedCallsShardPaths.get(shardIndex);
+                final File denoisedCopyRatioFile = getSampleDenoisedCopyRatioFile(shardRootDirectory, sampleIndex);
+                final DenoisedCopyRatioCollection shardDenoisedCopyRatioCollectionForShard = new DenoisedCopyRatioCollection(denoisedCopyRatioFile);
+                final List<SimpleInterval> shardIntervals = sortedIntervalCollections.get(shardIndex).getIntervals();
+                final String sampleNameFromDenoisedCopyRatioFile = shardDenoisedCopyRatioCollectionForShard
+                        .getMetadata().getSampleName();
+                Utils.validate(sampleNameFromDenoisedCopyRatioFile.equals(sampleName),
+                        String.format("Sample name found in the header of denoised copy ratio file for shard %d " +
+                                        "different from the expected sample name (found: %s, expected: %s).",
+                                shardIndex, sampleNameFromDenoisedCopyRatioFile, sampleName));
+                final List<LinearNonLocatableCopyRatio> shardDenoisedCopyRatioRecords = shardDenoisedCopyRatioCollectionForShard.getRecords();
+                Utils.validate(shardIntervals.size() == shardDenoisedCopyRatioRecords.size(),
+                        String.format("The number of entries in denoised copy ratio file for shard %d does " +
+                                        "not match the number of entries in the shard interval list (copy ratio list size: %d, " +
+                                        "interval list size: %d)", shardIndex, shardDenoisedCopyRatioRecords.size(),
+                                shardIntervals.size()));
+                concatenatedIntervalList.addAll(shardIntervals);
+                concatenatedDenoisedCopyRatioRecordsList.addAll(shardDenoisedCopyRatioRecords);
+            }
+            /* Attach the corresponding intervals */
+            final List<DenoisedLocatableCopyRatio> denoisedLocatableCopyRatioList =
```

Comment:
Maybe indent as follows to make this a bit more readable:

```            
            final List<DenoisedLocatableCopyRatio> denoisedLocatableCopyRatioList =
                    IntStream.range(0, concatenatedIntervalList.size())
                            .mapToObj(intervalIndex -> new DenoisedLocatableCopyRatio(
                                    concatenatedIntervalList.get(intervalIndex),
                                    concatenatedDenoisedCopyRatioRecordsList.get(intervalIndex)))
                            .collect(Collectors.toList());
```

Reply:
Done